### PR TITLE
mgr/dashboard: RBD Image Purge Trash, Move to Trash and Restore

### PR DESCRIPTION
**Pull request by rafaelquint**
Fixes: https://tracker.ceph.com/issues/40922
Fixes: https://tracker.ceph.com/issues/40928

Signed-off-by: Adam King <adking@redhat.com>
Signed-off-by: Rafael Quintero <rquinter@redhat.com>

### CODE REVIEW

**Comment by rafaelquint** [comment]
Has functionality from https://github.com/ceph/ceph/pull/29381 that were needed to create these tests.

**Comment by rafaelquint** [comment]
jenkins test dashboard

**Comment by rafaelquint** [comment]
jenkins test dashboard

**Comment by rafaelquint** [comment]
jenkins test dashboard

**Comment by rafaelquint** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/frontend/e2e/block/images.po.ts [hunk 1]
```diff
@@ -89,4 +90,119 @@ export class ImagesPageHelper extends PageHelper {
...
+    // wait for image to be created
+    browser.wait(
+      Helper.EC.not(Helper.EC.textToBePresentInElement($('.datatable-body'), '(Creating...)')),
+      Helper.TIMEOUT
+    );
+    return browser
+      .wait(Helper.EC.elementToBeClickable(this.getTableCell(name)), Helper.TIMEOUT)
+      .then(() => {
+        return this.getTableCell(name)
+          .click()
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/e2e/block/images.po.ts]:
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/e2e/block/images.po.ts [hunk 2]
```diff
@@ -89,4 +90,119 @@ export class ImagesPageHelper extends PageHelper {
...
+    this.navigateTo();
+    // wait for image to be created
+    browser.wait(
+      Helper.EC.not(Helper.EC.textToBePresentInElement($('.datatable-body'), '(Creating...)')),
+      Helper.TIMEOUT
+    );
+    return browser
+      .wait(Helper.EC.elementToBeClickable(this.getTableCell(name)), Helper.TIMEOUT)
+      .then(() => {
+        return this.getTableCell(name)
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/e2e/block/images.po.ts]:
```suggestion
        return this.moveClick(this.getTableCell(name))
```

File: src/pybind/mgr/dashboard/frontend/e2e/block/images.po.ts [hunk 3]
```diff
@@ -89,4 +90,119 @@ export class ImagesPageHelper extends PageHelper {
...
+
+  // Checks trash tab table for image and then restores it to the RBD Images table
+  // (could change name if new name is given)
+  restoreImage(name, newName?: string) {
+    this.navigateTo();
+    // clicks on trash tab
+    this.moveClick(element(by.cssContainingText('.nav-link', 'Trash'))).then(() => {
+      // wait for table to load
+      browser.wait(Helper.EC.elementToBeClickable(this.getTableCell(name)), Helper.TIMEOUT);
+      this.getTableCell(name).click();
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/e2e/block/images.po.ts]:
```suggestion
      this.moveClick(this.getTableCell(name));
```

**Comment by adk3798** [comment]
Rebased, fixed merge conflict and updated to fit new format using await.

Addressed some changes recommended by @tspmelo but not all since the moveClick function was removed in #29758 and the clickCheckbox function that replaced it only works with check boxes and labels.

**Comment by adk3798** [comment]
jenkins test make check

**Pull request merged**